exploration/number-selection.md

```diff
@@ -92,16 +95,78 @@ ICU MF1 messages using `plural` and `selectordinal` should be representable in M
-Using "plural" for "exact" still selects exactly matching cases,
+Number selection has three modes:
+- `exact` selection matches the operand to explicit numeric keys exactly
+- `plural` selection matches the operand to explicit numeric keys exactly
+  or to plural rule categories if there is no explicit match
+- `ordinal` selection matches the operand to explicit numeric keys exactly
+  or to ordinal rule categories if there is no explicit match
+
+The default selector function for numeric types or values is called `:number`.
+
+The operand of `:number` is either a literal that matches the `number-literal`
+production in the [ABNF](/main/spec/message.abnf) and which is parsed by the 
+implementation into a number; or an implementation-defined numeric type; 
+or is some serialization of a number that the implementation recognizes.
+
+> For example, in Java, any subclass of `java.lang.Number` plus the primitive
+> types (`boolean`, `byte`, `short`, `int`, `long`, `float`, `double`, etc.) 
+> might be considered as the "implementation-defined numeric types".
+> Implementations in other programming languages would define different types
+> or classes according to their local needs.
+
+Implementations SHOULD treat numeric literals that produce parse errors as
+non-numeric string literals.
+That is, numeric selection MUST emit a _Selection Error_ for these values.
+
+The function `:number` has the following options:
+- `select` (`plural`, `ordinal`, `exact`; default: `plural`)
+- `compactDisplay` (`short`, `long`; default: `short`)
+- `currency` (ISO 4712 currency code)
+- `currencyDisplay` (`symbol` `narrowSymbol` `code` `name`; default: `symbol`)
+- `currencySign` (`accounting`, `standard`; default: `standard`)
+- `notation` (`standard` `scientific` `engineering` `compact`; default: `standard`)
+- `numberingSystem` (arab arabext bali beng deva fullwide gujr guru hanidec khmr knda laoo latn 
+   limb mlym mong mymr orya tamldec telu thai tibt)
+- `signDisplay` (`auto` `always` `exceptZero` `never`; default=`auto`)
+- `style` (`decimal` `currency` `percent` `unit`; default=`decimal`)
+- `unit` (anything not empty)
+- `unitDisplay` (`long` `short` `narrow`; default=`short`)
+- `minimumIntegerDigits`, (positive integer, default: `1`)
+- `minimumFractionDigits`, (positive integer)
+- `maximumFractionDigits`, (positive integer)
+- `minimumSignificantDigits`, (positive integer, default: `1`)
+- `maximumSignificantDigits`, (positive integer, default: `21`)
+
+> [!NOTE]
+> We have a requirement for `offset` but have not provided for it yet??
+
+### Selection
+
+When performing selection, the resolved value of the `operand` is compared to 
+each of the variant keys.
+The result of such a comparison can be one of the following:
+
+- If the value of the key does not match the production `number-literal`
+  or is not one of the plural/ordinal keywords, the key is invalid;
+  return a _Selection Error_.
```

Comment:
This should specify the fallback behaviour, so probably we should emit the error and then ignore the invalid key in further processing?

Reply:
```
.match {$num :number}
oen  {{I ain't nothing but one misspelled}}
*    {{Should this fire instead?}}
```

Does "ignore the key in further processing" mean that the above selects `*`? Or does it trigger the fallback?

Reply:
With the above message, the `*` variant is always selected.

Reply:
Yes. I guess the question is whether `oen` should produce an error or just not match. My text says it's an error. I could imagine creating a sub-species of Selection Error for bad key values?